types/ember/test/object.ts

```diff
@@ -49,3 +49,20 @@ class Foo extends Ember.Object.extend({
         });
     }
 }
+
+export class Foo2 extends Ember.Object {
+  name!: string;
+
+  changeName(name: string) {
+    let a: string = this.set('name', name);
+    let b: number = this.set('name', name); // $ExpectError
+    let x: string = Ember.set(this, 'name', name);
+    let y: number = Ember.set(this, 'name', name); // $ExpectError
```

Comment:
This would be much cleaner w/ `// $ExpectType string`, but for some reason (possibly related to https://github.com/Microsoft/TypeScript/issues/27014 ) the return type is `this['name']`, which is assignable to `string`